File: Creational/Builder/PHP/interfaces/IBuilder.php
<?php

require_once('IEngine.php');

/**
 * Builder interface.
 */

interface IBuilder
{
	public function reset();

	/**
	 * @param string $engine
	 */
	public function setEngine(IEngine $engine);

	/**
	 * @param int $seats
	 */
	public function setSeats($seats);

	/**
	 * @param boolean $tComputer
	 */
	public function setTripComputer($tComputer);
}

File: Creational/Builder/Java/interfaces/IBuilder.java
package patterns.creational.builder.interfaces;

/**
 * Builder interface.
 */

public interface IBuilder {
    void reset();

    /**
     * @param engine
     */
    void setEngine(IEngine engine);

    /**
     * @param seats
     */
    void setSeats(int seats);

    /**
     * @param tComputer
     */
    void setTripComputer(boolean tComputer);
}
